numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -1075,6 +1075,9 @@ typedef void (NpyIter_GetMultiIndexFunc)(NpyIter *iter,
 #define NPY_ITER_PER_OP_FLAGS               0xffff0000
 
 
+#define NPY_ITER_FORCED_BROADCAST_AXIS(axis) (axis + (NPY_MAX_INT >> 1))
```

Comment:
This is a public header, we have to decide whether we think this is OK. Or move it to a private one for now. (That will be my default if nobody says anything. But to be clear. My feeling is that this is the much better API. As we can see based on all the weird behaviour, the general reduce OK flag does not really make much sense!)

Reply:
A good alternative may be to make this public as a function, not macro, so that it can be used but we are free to change how the flag looks like.

Reply:
Just to note, some things went wrong in the rebase probably... Or there were some bugs and I did not realize it... in any case some more fixup needed here.

Reply:
Making it private makes sense.  Function is OK too, but I'd go for minimum work...

Reply:
Yeah, will just move it to a private header later... Right now I got something lost with the broadcast axis being length 1 and that being lost...

Reply:
Moved it to `common.h` for now, should be OK although does not feel right. I have used `defined(NPY_INTERNAL_BUILD)` before, but that does not really seem right either?

In any case, its moved. And I guess there was a silly mistake in the last commit. The last commit is some general code duplication reduction and I could remove it, or reorder it so that the diff is possibly a bit simpler.

Unfortunately the whole giant reduce cleanup did not seem to be easy to disentangle :/...

Reply:
(sorry for the noise, this rebasing is going wrong, I will have to finish it tomorrow... including finishing the renaming)